### PR TITLE
C++ Linear Damper Tutorial

### PR DESCRIPTION
**Pull request by andermi**
tutorial: [CppLinearDamperExample](https://osrf.github.io/mbari_wec/pr-preview/pr-43/Tutorials/ROS2/CppLinearDamperExample/)

example code: [linear_damper_example](https://github.com/mbari-org/mbari_wec_template_cpp/tree/linear_damper_example)

### CODE REVIEW

File: docs/docs/Tutorials/ROS2/CppLinearDamperExample.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+`rpm`, `scale`, and `retract` are present in `buoy_interfaces.msg.PCRecord` on the `/power_data`
+topic published by the Power Controller running on the buoy.
+
+To access the data, all that is required is to define the callback
+`void Controller::power_callback(const buoy_interfaces::msg::PCRecord & data)`
+in the `Controller` class, and pass the data to `this->policy_->target` to get the desired winding
+current command. Various commands are available, and this time we will be using  
+`this->send_pc_wind_curr_command(wind_curr_amps);`
+
+``` cpp linenums="54" title="src/controller.cpp"
```
Comment (quarkytale) [inline docs/docs/Tutorials/ROS2/CppLinearDamperExample.md:253]:
```suggestion
``` cpp linenums="76" title="src/controller.cpp"
```

File: docs/docs/Tutorials/ROS2/PythonLinearDamperExample.md [hunk 2]
```diff
@@ -240,13 +240,21 @@ names along with other changes according to the previous tutorial.
...
+$ source install/local_setup.bash
+```
+
 We will be using `ros2 launch` and `launch/controller.launch.py` to run our new controller.
 
-To run the controller along with the simulation, first source your workspace. Then, launch your
+To run the controller along with the simulation, launch your
 controller:  
-`$ ros2 launch mbari_wec_linear_damper_py controller.launch.py`
+`$ ros2 launch mbari_wec_linear_damper_cpp controller.launch.py`
```
Comment (quarkytale) [inline docs/docs/Tutorials/ROS2/PythonLinearDamperExample.md:255]: Umm this should be `mbari_wec_linear_damper_py`

Comment (andermi) [reply docs/docs/Tutorials/ROS2/PythonLinearDamperExample.md:255]: I'm not sure what you mean here.

Comment (mabelzhang) [reply docs/docs/Tutorials/ROS2/PythonLinearDamperExample.md:255]: Maybe she thought it's a typo, based on the file extension? 😅 It's this file https://github.com/mbari-org/mbari_wec_template_cpp/blob/main/launch/controller.launch.py

**Comment by mabelzhang** [comment]
General comment on these tutorials - I've been noticing that at the top, the navigation goes from Home (an icon) directly to the individual "Quick Start -- ..." tutorials. Then if the user clicks on Home, they can't find how to go to other tutorials.

I think we need to add a navigation list for the tutorials to this YAML file, maybe under the `Tutorials` line:
https://github.com/osrf/mbari_wec/blob/f53a957cdf5869df81cd3681a05280bf05d490b0/docs/mkdocs.yml#L10
one bullet per tutorial, so that the navigation bar will be like Home > Tutorials > "Quick Start -- ..."

See https://www.mkdocs.org/user-guide/writing-your-docs/#configure-pages-and-navigation

This can be in a new PR.

File: docs/docs/Tutorials/ROS2/CppLinearDamperExample.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+- `torque_constant`: Motor Torque Constant (N-m/Amp)  
+  Constant to convert desired torque to applied motor winding current
+<br/>
+<br/>
+- `n_spec`: Input Motor Speed (RPM) Breakpoints  
+  \(N\) (RPM) is the input to the controller and `n_spec` are the x-components of the breakpoints
+  \(\left(n\_spec, \frac{torque\_spec}{torque_constant}\right)\) for the interpolant,  
+<br/>
+  \(\hat{f}_{I}(n\_spec) = \frac{torque\_spec}{torque\_constant} \approx f_{I}(N) = I\)
```
Comment (mabelzhang) [inline docs/docs/Tutorials/ROS2/CppLinearDamperExample.md:64]: Is the hat on f supposed to be over both f and I? Looks funny.

File: docs/docs/Tutorials/ROS2/CppLinearDamperExample.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+```
+
+So, as you can see we apply a positive damping torque when RPM is negative (piston extending), and
+a positive damping torque when RPM is positive (piston retracting). The damping torque required is
+reduced when retracting.
+
+### Controller
+
+All that is left is to connect the necessary feedback data to the `ControlPolicy`. In this case,
+`rpm`, `scale`, and `retract` are present in `buoy_interfaces.msg.PCRecord` on the `/power_data`
```
Comment (mabelzhang) [inline docs/docs/Tutorials/ROS2/CppLinearDamperExample.md:244]: Can we link to the message file?
```suggestion
`rpm`, `scale`, and `retract` are present in [`buoy_interfaces.msg.PCRecord`](https://github.com/osrf/mbari_wec_utils/blob/main/buoy_interfaces/msg/PCRecord.msg) on the `/power_data`
```

**Review by mabelzhang — COMMENTED** [review]
Very nice. I'm impressed there are no English typos in this long page 😃

**Review by mabelzhang — APPROVED** [review]
Hmm since my review was requested - looks good to me, but I didn't read the code explanation closely - I think @quarkytale did that. My review was a high-level conceptual read and compiling and running it. Runs for me, buoy moves a bit faster than when it's not running, so I guess it works.

**Pull request merged**